src/Core/Hook/Provider/GridDefinitionHookByServiceIdsProvider.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Hook\Provider;
+
+use Generator;
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\GridDefinitionFactoryInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides hooks list by calling service ids from the container.
+ */
+final class GridDefinitionHookByServiceIdsProvider implements HookByServiceIdsProviderInterface
```

Comment:
I think this service should be inside bundle, @matks wdyt?

Reply:
Nope => https://devdocs.prestashop.com/1.7/development/architecture/understanding-src-folder/

Reply:
But this class **depends on container**, why would it be inside `Core`?

Reply:
Answered later:

> I'm wondering why you don't populate a parameter at container builder "building" using a Compiler Pass instead: would be more performant.